src/include/wtfutex.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+
+/*
+ * Wrap the futex (address of 32-bit value) in a structure to signify
+ * that the value should not be modified directly.
+ */
+struct __wt_futex {
+  _Alignas(4) uint32_t 
+};
+
+
+/*
+ * Restricted for portability.
+ */
+enum __wt_futex_wake {
+  WT_WAKE_ONE,			/* Wake a single waiting thread. */
+  WT_WAKE_ALL			/* Wake all waiting threads. */
+};
+
+
+/*
+ * Get the value associated with the futex.
+ *
+ * \retval -EINVAL Not a futex, or pointer to [val] is null.
+ */
+int __wt_fetch(struct __wt_futex *ftx, uint32_t *val);
```

Comment:
Is there a reason not to call this __wt_futex_fetch? I'm trying to figure out if fetch/store consciously skip the `futex` prefix.

I'm not that familiar with futexes, but the interface seems sensible to me as is.

Reply:
No: entirely inadvertent. Fixed -- thanks.